### PR TITLE
Remove fetch-canonical-data-syncer scripts

### PR DESCRIPTION
**Pull request by ErikSchierboom**
The `fetch-canonical-data-syncer` script is obsolete as canonical data syncing is now built into `configlet`.

See https://github.com/exercism/configlet/issues/124

### CODE REVIEW

File: bin/check_test_cases [hunk 1]
```diff
@@ -4,7 +4,5 @@ die() { echo "$*" >&2; exit 1; }
 cd "$(dirname "$0")"/.. || die "cannot cd"
 
 bin/fetch-configlet || die "cannot fetch configlet"
-bin/configlet lint . || die "resolve config.json problems first"
-
-bin/fetch-canonical_data_syncer || die "cannot fetch canonical_data_syncer"
-bin/canonical_data_syncer --check "$@"
+bin/configlet lint || die "resolve config.json problems first"
+bin/configlet sync --check || die "resolve tests.toml problems first"
```
Comment (ee7) [inline bin/check_test_cases:8]:
```suggestion
bin/configlet sync --check "$@"
```

Maybe.

But the previous final line in this script was
```
bin/canonical_data_syncer --check "$@"
```

**Review by ee7 — COMMENTED** [review]
I've checked that, with this PR,
```
$ git grep 'syncer'
```
produces no output.

File: bin/check_test_cases (hunk 1 above)
Comment (ErikSchierboom) [reply bin/check_test_cases:8]: 🤷

Comment (glennj) [reply bin/check_test_cases:8]: I did that so I could check all exercises, or pass an `--exercise slug` option for only that one,

**Review by glennj — APPROVED** [review]

**Pull request merged**